### PR TITLE
Remove duplication of nav markup

### PR DESCRIPTION
**Pull request by Volker-E**
Removing duplicated navigation markup and fix responsive menu trigger.
Also:
  - Improving HTML header markup by reducing it largely
  - Adding `title` and `aria-hidden` attributes to several header
    elements
  - Fixing incorrect ARIA roles to `role="navigation"`
  - Removing Bootstrap grid and replace with minimal custom one
  - Fixing hamburger icon visually and functionally
  - Improving responsive layout and behaviour by reducing UI elements
    to the bare minimum on smaller devices
  - Integrating `transition`s on various elements for a smoother user
    experience

### CODE REVIEW

File: css/wmui-style-guide.dev.css [hunk 1]
```diff
@@ -286,80 +249,94 @@ code {
 	margin-top: 6.4rem;
 	padding: 6.4rem 0;
 	box-shadow: 0 1px 4px 0 rgba( 0, 0, 0, 0.25 );
+	overflow-y: hidden;
```
Comment (juliengirault) [inline css/wmui-style-guide.dev.css]: 🤔

File: css/wmui-style-guide.dev.css [hunk 2]
```diff
@@ -286,80 +249,94 @@ code {
...
 }
-.toc ol:first-child {
+.nav--main ol:first-child {
 	margin: 0;
 }
 
-.toc li {
+.nav--main li {
+	padding: 0;
 	list-style: none none;
```
Comment (juliengirault) [inline css/wmui-style-guide.dev.css]: why `none none` ?

File: css/wmui-style-guide.dev.css [hunk 3]
```diff
@@ -462,54 +464,145 @@ code {
...
+		padding-left: 1.6rem;
+	}
+
+	figure {
+		margin-left: 0;
+		margin-right: 0;
+	}
 }
 
 /* Portrait tablet to landscape and desktop */
```
Comment (juliengirault) [inline css/wmui-style-guide.dev.css]: Below there is `.box` but I couldn't find any use in the html

File: design-principles.html [hunk 4]
```diff
@@ -21,56 +21,31 @@
...
-							<li><a href="#imagery">Imagery</a></li>
-							<li><a href="#illustrations">Illustrations</a></li>
-						</ul>
-					</li>
-					<!-- <li><a href="components.html">Components</a></li>
-					<li><a href="patterns.html">Patterns</a></li>
-					<li><a href="resources.html">Resources</a></li> -->
-				</ol>
-			</nav>
+			<a href="https://github.com/wikimedia/WikimediaUI-Style-Guide" class="lnk--contribute" title="Fork and contribute on GitHub"><span>Contribute on </span>Github</a>
```
Comment (Volker-E) [inline design-principles.html]: Done.

File: visual-style.html [hunk 5]
```diff
@@ -21,56 +21,31 @@
...
 	</header>
 	<div class="page">
 		<div class="content-box">
-			<div class="col-xs-hidden col-sm-4 col-md-3">
-				<nav id="nav" class="nav toc" role="nav">
+			<div class="col col--start">
+				<input type="checkbox" id="trigger--nav-main" class="trigger--nav-main">
+				<nav id="nav--main" class="nav nav--main" role="navigation">
 					<ol>
 						<li><a href="index.html">Introduction</a></li>
```
Comment (Volker-E) [inline visual-style.html]: Oh yes (not connected to any treatment, therefore under my radar). Thanks. Done.

File: css/wmui-style-guide.dev.css (hunk 1 above)
Comment (Volker-E) [inline css/wmui-style-guide.dev.css]: That's for the navigation `transform: translateY()`.

File: css/wmui-style-guide.dev.css (hunk 2 above)
Comment (Volker-E) [inline css/wmui-style-guide.dev.css]: `list-style: none none` is a way to make old IE folow it by applying
- `list-style-image: none`
- `list-style-type: none`

File: css/wmui-style-guide.dev.css (hunk 3 above)
Comment (Volker-E) [inline css/wmui-style-guide.dev.css]: Good catch, done.

**Comment by Volker-E** [comment]
Also fixed navigation `is-on` falsely set to bold and slightly adapted the vertical alignment of logo further. The logo position has changed in order to be aligned with hamburger and “Contribute” link.

File: css/wmui-style-guide.dev.css (hunk 2 above)
Comment (juliengirault) [inline css/wmui-style-guide.dev.css]: Ok, but it gets removed when minified.

File: css/wmui-style-guide.dev.css (hunk 1 above)
Comment (juliengirault) [inline css/wmui-style-guide.dev.css]: 👍

**Pull request merged**